server/src/main/java/timely/Configuration.java

```diff
@@ -1,196 +1,608 @@
+        }
+
+        public Ssl getSsl() {
+            return ssl;
+        }
+    }
+
+    @NotEmptyIfFieldSet.List({
+            @NotEmptyIfFieldSet(fieldName = "useGeneratedKeypair", fieldValue = "false", notNullFieldName = "certificateFile", message = "must be set if timely.security.ssl.use-generated-keypair is false"),
+            @NotEmptyIfFieldSet(fieldName = "useGeneratedKeypair", fieldValue = "false", notNullFieldName = "keyFile", message = "must be set if timely.security.ssl.use-generated-keypair is false") })
+    public class Ssl {
+
+        private String certificateFile;
+        private String keyFile;
+        private String keyPassword;
+        private String trustStoreFile;
+        private boolean useGeneratedKeypair = false;
+        private boolean useOpenssl = true;
+        private List<String> useCiphers = Lists.newArrayList("TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+                "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_128_GCM_SHA256",
+                "TLS_RSA_WITH_AES_128_CBC_SHA", "SSL_RSA_WITH_3DES_EDE_CBC_SHA");
+
+        public String getCertificateFile() {
+            return certificateFile;
+        }
+
+        public Configuration setCertificateFile(String certificateFile) {
+            this.certificateFile = certificateFile;
+            return Configuration.this;
+        }
+
+        public String getKeyFile() {
+            return keyFile;
+        }
+
+        public Configuration setKeyFile(String keyFile) {
+            this.keyFile = keyFile;
+            return Configuration.this;
+        }
+
+        public String getKeyPassword() {
+            return keyPassword;
+        }
+
+        public Configuration setKeyPassword(String keyPassword) {
+            this.keyPassword = keyPassword;
+            return Configuration.this;
+        }
+
+        public String getTrustStoreFile() {
+            return trustStoreFile;
+        }
+
+        public Configuration setTrustStoreFile(String trustStoreFile) {
+            this.trustStoreFile = trustStoreFile;
+            return Configuration.this;
+        }
+
+        public boolean isUseGeneratedKeypair() {
+            return useGeneratedKeypair;
+        }
+
+        public Configuration setUseGeneratedKeypair(boolean useGeneratedKeypair) {
+            this.useGeneratedKeypair = useGeneratedKeypair;
+            return Configuration.this;
+        }
+
+        public boolean isUseOpenssl() {
+            return useOpenssl;
+        }
+
+        public Configuration setUseOpenssl(boolean useOpenssl) {
+            this.useOpenssl = useOpenssl;
+            return Configuration.this;
+        }
+
+        public List<String> getUseCiphers() {
+            return useCiphers;
+        }
+
+        public Configuration setUseCiphers(List<String> useCiphers) {
+            this.useCiphers = useCiphers;
+            return Configuration.this;
+        }
+    }
+
+    public class Server {
+
+        @NotBlank
```

Comment:
I had meant to use @NotEmpty everywhere. However, I just read the doc and it says for @NotBlank: ```The difference to {@code NotEmpty} is that trailing whitespaces are getting ignored.```
So maybe I should be using @NotBlank everywhere.